atomic_reactor/plugins/post_tag_and_push.py

```diff
@@ -148,86 +148,82 @@ def run(self):
-                        manifests_dict = get_all_manifests(registry_image, registry, insecure,
-                                                           docker_push_secret, versions=('v2',))
-                        try:
-                            koji_source_manifest_response = manifests_dict['v2']
-                        except KeyError as exc:
-                            raise RuntimeError(
-                                f'Unable to fetch v2 schema 2 digest for {registry_image.to_str()}'
-                            ) from exc
-
-                        wf_data.koji_source_manifest = koji_source_manifest_response.json()
-
-                    digests = get_manifest_digests(registry_image, registry,
-                                                   insecure, docker_push_secret)
-
-                    if (not (digests.v2 or digests.oci) and (retry < max_retries)):
-                        sleep_time = DOCKER_PUSH_BACKOFF_FACTOR * (2 ** retry)
-                        self.log.info("Retrying push because V2 schema 2 or "
-                                      "OCI manifest not found in %is", sleep_time)
-
-                        time.sleep(sleep_time)
-                    else:
-                        break
-
-                pushed_images.append(registry_image)
-
-                tag = registry_image.to_str(registry=False)
-                push_conf_registry.digests[tag] = digests
-
-                if not config_manifest_digest and (digests.v2 or digests.oci):
-                    if digests.v2:
-                        config_manifest_digest = digests.v2
-                        config_manifest_type = 'v2'
-                    else:
-                        config_manifest_digest = digests.oci
-                        config_manifest_type = 'oci'
-                    config_registry_image = registry_image
+        insecure = self.registry.get('insecure', False)
+
+        docker_push_secret = self.registry.get('secret', None)
+        self.log.info("Registry %s secret %s", self.registry['uri'], docker_push_secret)
+
+        for image in wf_data.tag_conf.images:
+            if image.registry:
+                raise RuntimeError("Image name must not contain registry: %r" % image.registry)
+
+            if not source_docker_archive:
+                image_size = sum(item['size'] for item in self.workflow.data.layer_sizes)
+                config_image_size = image_size_limit['binary_image']
+                # Only handle the case when size is set > 0 in config
+                if config_image_size and image_size > config_image_size:
+                    raise ExceedsImageSizeError(
+                        'The size {} of image {} exceeds the limitation {} '
+                        'configured in reactor config.'
+                        .format(image_size, image, image_size_limit)
+                    )
+
+            registry_image = image.copy()
+            registry_image.registry = self.registry['uri']
```

Comment:
here as well

Reply:
this one is a bit confusing. the registry is being set to `None` at https://github.com/containerbuildsystem/atomic-reactor/blob/master/atomic_reactor/plugins/post_tag_and_push.py#L128 so I'm not sure if removing it here is the best idea

Reply:
'here as well' what?

Reply:
you should just add registry to those additional tags 

        if not tag_conf.unique_images:
            if source_docker_archive:
                tag_conf.add_unique_image(source_unique_image)
            else:
                tag_conf.add_unique_image(self.workflow.image)

as the rest will have registry already from pre_tag_from_config

Reply:
also additional idea is to have new method in TagConf 'add_registry'
and before you use tag_conf for reading you will just force 'add' registry there in bulk to be sure ...

Reply:
or even better you could just call that add_registry once, to set registry for whole TagConf

and when you call images/floating_images etc, it will already return those with added registry

that way you won't have to add it in plugins

Reply:
registry is already there but it is being removed for some reason and then added back to the copy of the image. That's what is confusing me, if it's being done for a particular reason or not

Reply:
it is being removed just for source image, which name is generated from that binary build, and in koji there can be some old registry, so that is why it is removed

Reply:
updated it to set the registry only for source image, for binary images it is left as as. let me know if it's good